### DISCUSSION TITLE
Get Access to YAML metadata using Plugin, and possibly filter out some APIs using C# code and generate a CSV format.

### DISCUSSION THREAD

**Discussion by chaitanyamalik** [dotnet/docfx#8162] [General]
Hey!

I'm relatively new in .NET and C# coding and it's the first time I came across DocFX.

I'm just dealing with metadata for a project and I don't need the HTML content in any way. I want to extract some content for unit tests in my project which is being extracted well by `docfx metadata` into YAML format.

I want to be able to filter out only the APIs I need. I tried that using filter_config but it didn't seem to workout. I raised an issue for the same. https://github.com/dotnet/docfx/issues/8150

However, I've been told that a better use and implementation of my task would be, if I am able to access YAML files using Plugins. 

The documentation on the usage of Plugins is quite thin, so here I am looking for some guidance.

My goal is to load a list of Unit Tests and certain key value pairs of that test from YAML that match a certain condition, into a CSV format.

Is this possible to do just by writing a block of code in C# using DocFX Plugins..?

**Comment by KalleOlaviNiemitalo** [comment]
I suspect Roslyn APIs would be better for this than DocFX. You'd be able to support new language versions without waiting for DocFX v2 to be updated.

**Reply by chaitanyamalik** [reply]
@KalleOlaviNiemitalo Can you help me understand how the Roslyn API comes into play in my case..?

For now, DocFX generates all the metadata for a unit test in a YAML format, written in a .cs file. In my case, the relevant information I want to gather is the summary (/// comments), method/test name, its path and its attributes...

As of now I'm having to parse multiple YAML files with large amounts of unnecessary data, to filter out the ones I want, and thus convert them in a CSV format afterwards. But this is a step I'm looking to avoid.

How can I do this in Roslyn?

On a separate note, Is this not achievable in DocFX yet? (Accessing what goes into the YAML using Plugin)

**Reply by KalleOlaviNiemitalo** [reply]
DocFX v2 is not very actively maintained nowadays (issues about supporting new versions of C# remain open for a long time), and AFAIK DocFX v3 does not support loading plugins, so I wouldn't recommend implementing a plugin for DocFX.

**Reply by paulushub** [reply]
> How can I do this in Roslyn?

Roslyn is the library used by DocFX for this and more. See the `Microsoft.DocAsCode.Metadata.ManagedReference.Roslyn`.
Better still start with Roslyn tutorials like this one [Roslyn Succinctly](https://www.syncfusion.com/succinctly-free-ebooks/roslyn).
